source/extensions/tracers/xray/tracer.cc

```diff
@@ -76,17 +77,28 @@ void Span::finishSpan() {
   s.set_end_time(
       time_point_cast<SecondsWithFraction>(time_source_.systemTime()).time_since_epoch().count());
   s.set_parent_id(parentId());
-  using KeyValue = Protobuf::Map<std::string, std::string>::value_type;
-  for (const auto& item : custom_annotations_) {
-    s.mutable_annotations()->insert(KeyValue{item.first, item.second});
+
+  // HTTP annotations
+  using StructField = Protobuf::MapPair<std::string, ProtobufWkt::Value>;
+  daemon::Segment_http_annotations* http = s.mutable_http();
+
+  ProtobufWkt::Struct* request = http->mutable_request();
```

Comment:
I'd change those two lines to:
```cpp
ProtobufWkt::Struct* request = s.http()->mutable_request();
```
The `http` is not used anywhere else.

Reply:
makes sense. this was left over from my misunderstanding that constructing submessages and assigning isn't the right way to go in c++.